db/db_impl/db_impl.cc

```diff
@@ -3720,6 +3721,31 @@ ArenaWrappedDBIter* DBImpl::NewIteratorImpl(
   return db_iter;
 }
 
+std::unique_ptr<MultiCfIterator> DBImpl::NewMultiCfIterator(
+    const ReadOptions& _read_options,
+    const std::vector<ColumnFamilyHandle*>& column_families) {
+  if (column_families.size() < 2) {
+    return std::make_unique<EmptyMultiCfIterator>(
+        Status::InvalidArgument("Less than 2 CFs were provided"));
```

Comment:
Hmm, what to do with a single CF is a tricky question. I initially though that InvalidArgument was a good idea to promote efficient use of the best API, but then I remembered that checking status from a returned iterator is not always done properly. It would be easy for a user to mix up the result provided here with "success but no data found," if they only check valid() and not status(). We don't encourage sloppy ignoring of Statuses, but we should not ignore that it happens (hahahaha).

Another possibility to consider is to just make multi-CF awareness part of plain Iterator and then NewMultiCfIterator() with a single CF could be exactly equivalent to NewIterator(), with a special case in this code.

Do we foresee adding things to the abstract MultiCfIterator that would be difficult or nonsensical to implement for a single-CF iterator?